diff --git a/lib/cli/vite-dev.js b/lib/cli/vite-dev.js
index cd8a6da777c600bfd13715f5585bb183891adb59..c4bd093b2147aef5fb433b5544718b880f620ff5 100644
--- a/lib/cli/vite-dev.js
+++ b/lib/cli/vite-dev.js
@@ -37,6 +37,10 @@ const bundler = async (config, configFolder) => {
         middlewareMode: true,
       },
     });
+    // Support vite-plugin-react-swc https://github.com/tajo/ladle/issues/294#issuecomment-1405265935
+    if (viteConfig.plugins[0].some(({name}) => name === 'vite:react-swc')) {
+       viteConfig.plugins[0] = (await import('@vitejs/plugin-react')).default();
+     }
     const vite = await createServer(viteConfig);
     const { moduleGraph, ws } = vite;
     app.head("*", async (_, res) => res.sendStatus(200));
